clang/lib/AST/ASTImporter.cpp

```diff
@@ -3165,6 +3165,7 @@ ExpectedDecl ASTNodeImporter::VisitRecordDecl(RecordDecl *D) {
                 if (Error Err = ImportImplicitMethods(DCXX, FoundCXX))
                   return std::move(Err);
             }
+            return FoundDef;
```

Comment:
I put this probably into a new pull request or omit this change. But in the current code I do not see any reason why we can not return here. If there is no return, the later `GetIimportedOrCreateDecl` call should return the existing object. But the imports that are done before the call are not necessary.

Reply:
So currently what happens is that we create a new decl and connect it to `FoundDef` on a redeclaration chain? LLDB is particularly sensitive to this particular code-path, so I'd say it's best to try do this in a separate PR so we can focus in on this

Reply:
The later `GetImportedOrCreateDecl` returns the "already imported" declaration that was set by the `MapImported` (line 3157) call. So in this case we should return on line 3227.